### PR TITLE
Squad with Horovod, and multiple stage training scripts

### PR DESCRIPTION
**Pull request by c0redumb**
Added Horovod to Squad downstream training. Fix incompatibilities after v2 conversion.

Added scripts to do multiple stage training (multi-stage = different sequence length / batch size adjustable in script)

### CODE REVIEW

File: run_squad.py [hunk 1]
```diff
@@ -154,10 +159,13 @@
     "null_score_diff_threshold", 0.0,
     "If null_score - best_non_null is greater than the threshold predict null.")
 
-flags.DEFINE_string("optimizer_type", "adam", "Optimizer used for training - adam (default), lamb, nadam and nlamb")
+flags.DEFINE_string(
+    "optimizer_type", "adam", "Optimizer used for training - adam (default), lamb, nadam and nlamb")
```
Comment (micmelesse) [inline run_squad.py:163]: unnecessary change

File: modeling.py [hunk 2]
```diff
@@ -338,7 +338,8 @@ def get_assignment_map_from_checkpoint(tvars, init_checkpoint):
     (name, var) = (x[0], x[1])
     if name not in name_to_variable:
       continue
-    assignment_map[name] = name
+    #assignment_map[name] = name
+    assignment_map[name] = name_to_variable[name]
```
Comment (micmelesse) [inline modeling.py:342]: unnecessary comment

File: run_squad.py (hunk 1 above)
Comment (c0redumb) [reply run_squad.py:163]: Thanks a lot, Michael. That's probably from the autoformatter because the line is too long.

Maybe we want to keep the line that way for now. I don't know how to pick and choose within a commit.

File: modeling.py (hunk 2 above)
Comment (c0redumb) [reply modeling.py:342]: For non-trivial changes, I usually keep the old code in comment as reference. This can be removed later when all goes well.

This particular change is to deal with a problem that a perfectly running code (checkpoint loading) suddenly stopped to work for TF v2. I still haven't yet got time to fully understand why it failed for TF v2, and why that changed line makes it work. So I am keeping that old line in comment, and will come back to this later.
